File: client/protocol/messages.py
from dataclasses import dataclass

@dataclass
class Message:
    type: str
    request_id: str
    payload: dict

@dataclass
class RegisterPayload:
    subdomain: str

@dataclass
class RegisteredPayload:
    url: str

@dataclass
class HttpRequestPayload:
    method: str
    path: str
    headers: dict
    body: str

@dataclass
class HttpResponsePayload:
    status_code: int
    headers: dict
    body: str

@dataclass
class ErrorPayload:
    err: str

File: server/protocol/messages.go
package protocol

import "encoding/json"

type Message struct {
	Type      string
	RequestID string
	Payload   json.RawMessage
}

type RegisterPayload struct {
	Subdomain string
}

type RegisteredPayload struct {
	URL string
}

type HttpRequestPayload struct {
	Method  string
	Path    string
	Headers map[string]string
	Body    string
}

type HttpResponsePayload struct {
	StatusCode int
	Headers    map[string]string
	Body       string
}

type ErrorPayload struct {
	Error string
}
